css-overflow-5/Overview.bs

```diff
@@ -938,47 +936,51 @@ Issue: Explore whether scrolltarget can be more directly associated with anchor
 	or might directly track a user’s input
 	(e.g. touch scrolling, scrollbar dragging).
 	In either case, the user agent chooses an 'eventual scroll position' to which the scroller
-	will reach.
+	will reach. This ensures that the relevant marker is activated immediately.
 
-	This 'eventual scroll position' is used to determine the active marker.
+	This 'eventual scroll position' is used to determine the active marker within each [=scroll marker group=].
 	Since markers themselves may represent just the start of the content (e.g. headers), we consider the active marker to be the first one which we are at or beyond the scroll position of.
 
 	<div algorithm="scroll tracking">
 		Whenever a [=scroll container=] is scrolled, or layout changes the scroll position, the user agent must run these steps to determine the active marker:
 
 		1. Let <var>position</var> be the 'eventual scroll position' for the scrolling operation.
-		1. Let <var>markers</var> be all of the [=scroll marker control=] elements which are a part of the [=scroll marker group=] for the [=scroll container=].
-		1. Let <var>targets</var> be the set of {{HTMLButtonElement/scrollTargetElement}} for those controls sorted in [=tree order=].
-		1. Let <var>selected</var> be the first element in <var>targets</var>, or null if <var>targets</var> is empty.
-		1. For each <var>target</var> in <var>targets</var>:
-			1. Set <var>selected</var> to <var>target</var>.
-			1. Let <var>targetPosition</var> be the position that would be scrolled to if we scroll <var>target</var> into view.
-			1.	: If <var>targetPosition</var>'s scroll block offset is less than or equal to <var>position</var>'s scroll block offset, and
-					<var>targetPosition</var>'s scroll inline offset is less than or equal to <var>position</var>'s scroll inline offset.
+		1. For each focusgroup <var>group</var> containing one or more [=scroll marker control=] elements whose {{HTMLLinkElement/scrollTargetElement}} is a descendant of [=scroll container=]:
+
+			1. Let <var>markers</var> be all of the [=scroll marker control=] elements which are a part of the [=scroll marker group=] for the [=scroll container=].
+			1. Let <var>targets</var> be the set of {{HTMLLinkElement/scrollTargetElement}} for those controls sorted in [=tree order=].
+			1. Let <var>selected</var> be the first element in <var>targets</var>, or null if <var>targets</var> is empty.
+			1. For each <var>target</var> in <var>targets</var>:
+				1. Set <var>selected</var> to <var>target</var>.
+				1. Let <var>targetPosition</var> be the position that would be scrolled to if we scroll <var>target</var> into view.
+				1.	: If <var>targetPosition</var>'s scroll block offset is less than or equal to <var>position</var>'s scroll block offset, and
+						<var>targetPosition</var>'s scroll inline offset is less than or equal to <var>position</var>'s scroll inline offset.
+					::
+						Update <var>selected</var> to <var>target</var>.
+						Break.
```

Comment:
This is super algorithmic, to a degree that I think hampers readability rather then helps clarity. Can boil it down to:

> Let |markers| be ....
> 
> Let |targets| be the {{HTMLLinkElement/scrollTargetElement}}s of |markers|, associated with the item of |markers| they came from, and sorted in tree order.
> 
> For each item in |targets|, associate it with a |target position|, which is ...
> 
> Let |selected target| be the largest-indexed item of |targets| whose associated |target position| is ...
> If there is no such item, return.
> 
> Let |selected marker| be the marker associated with |selected target|.
> If multiple items of |markers| are associated with |selected target|,
> set |selected marker| to be the marker that is earliest in tree order.
> 
> ...

Reply:
Thanks for the suggestion. I've adapted the algorithm to follow roughly what you suggested here with the appropriate sections filled in.